src/test/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSourceConfigTest.java

```diff
@@ -0,0 +1,159 @@
+        assertEquals("https://example.org/example", actualSalesforceStreamingSourceConfig.getLoginUrl());
+        assertEquals("Push Topic Name", actualSalesforceStreamingSourceConfig.getPushTopicName());
+        assertSame(oAuthInfo, actualSalesforceStreamingSourceConfig.getOAuthInfo());
+        assertEquals("Consumer Secret", actualSalesforceStreamingSourceConfig.getConsumerSecret());
+        assertEquals("passwordtoken", actualSalesforceStreamingSourceConfig.getPassword());
+        assertEquals("username", actualSalesforceStreamingSourceConfig.getUsername());
+    }
+
+    @Test
+    public void testFetchPushTopicByNameInvalidFieldName()
+            throws ConnectionException {
+        PartnerConnection partnerConnection = mock(PartnerConnection.class);
+        try {
+            SalesforceStreamingSourceConfig.fetchPushTopicByName(partnerConnection, ",");
+        } catch (IllegalArgumentException e) {
+            assertEquals("Push topic name ',' can only contain latin letters.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void testFetchPushTopicByNameTestWhenZeroRecord()
+            throws ConnectionException {
+        SObject[] sObjectArray = {};
+        QueryResult queryResult = mock(QueryResult.class);
+        PartnerConnection partnerConnection = mock(PartnerConnection.class);
+        Mockito.when(partnerConnection.query(anyString())).thenReturn(queryResult);
+        Mockito.when(queryResult.getRecords()).thenReturn(sObjectArray);
+        assertEquals(null, SalesforceStreamingSourceConfig.fetchPushTopicByName(partnerConnection, "test"));
+    }
+
+    @Test
+    public void fetchPushTopicByNameTestWhenSingleRecord() throws ConnectionException {
+        SObject sObject = mock(SObject.class);
+        SObject[] sObjectArray = {sObject};
+        QueryResult queryResult = mock(QueryResult.class);
+        PartnerConnection partnerConnection = mock(PartnerConnection.class);
+        Mockito.when(partnerConnection.query(anyString())).thenReturn(queryResult);
+        Mockito.when(queryResult.getRecords()).thenReturn(sObjectArray);
+        assertEquals(sObject, SalesforceStreamingSourceConfig.fetchPushTopicByName(partnerConnection, "test"));
+    }
+
+    @Test
+    public void fetchPushTopicByNameTestWhenMultipleRecord() throws ConnectionException {
+        SObject sObject = mock(SObject.class);
+        SObject[] sObjectArray = {sObject, sObject};
+        QueryResult queryResult = mock(QueryResult.class);
+        PartnerConnection partnerConnection = mock(PartnerConnection.class);
+        Mockito.when(partnerConnection.query(anyString())).thenReturn(queryResult);
+        Mockito.when(queryResult.getRecords()).thenReturn(sObjectArray);
+        try {
+            SalesforceStreamingSourceConfig.fetchPushTopicByName(partnerConnection, "test");
+        } catch (IllegalStateException e) {
+            assertEquals("Excepted one or zero pushTopics with name = 'test' found 2", e.getMessage());
+        }
```

Comment:
fail if exception in not thrown

Reply:
Failure Assertion added, if no exception thrown